Services/Simulation/JavascriptInterpreter.cs

```diff
@@ -84,16 +93,21 @@ public class JavascriptInterpreter : IJavascriptInterpreter
                 }
 
                 this.log.Debug("Executing JS function", () => new { filename });
-                JsValue output = engine.Execute(program).Invoke("main", context, this.deviceState);
 
-                var result = this.JsValueToDictionary(output);
-                this.log.Debug("JS function success", () => new { filename, result });
-                return result;
+                JsValue output = engine.Execute(program).Invoke(
+                    "main",
+                    context,
+                    this.deviceState.GetAll(),
+                    this.deviceProperties.GetAll());
+
+                // update the internal device state with the new state
+                this.UpdateState(output);
```

Comment:
now there are 2 ways of updating the device state, either `main()` returns a state or it calls `updateState()`. Do you think we should drop the former?

Reply:
It feels more intuitive to me to return a state from main() rather than requiring the script to call updateState() before exiting. But certainly, from a functional perspective calling updateState() would serve the same purpose.